apis/python/src/tiledbsoma/_point_cloud.py

```diff
@@ -0,0 +1,255 @@
+# Copyright (c) 2024 The Chan Zuckerberg Initiative Foundation
+# Copyright (c) 2024 TileDB, Inc.
+#
+# Licensed under the MIT License.
+"""
+Implementation of a SOMA Point Cloud
+"""
+
+import warnings
+from typing import Any, Optional, Sequence, Tuple, Union
+
+import pyarrow as pa
+import somacore
+from somacore import CoordinateSpace, CoordinateTransform, options
+from typing_extensions import Self
+
+from ._constants import SOMA_JOINID, SPATIAL_DISCLAIMER
+from ._dataframe import Domain
+from ._read_iters import TableReadIter
+from ._types import OpenTimestamp
+from .options import SOMATileDBContext
+
+_UNBATCHED = options.BatchSize()
+
+
+class PointCloud(somacore.PointCloud):
```

Comment:
I do like `PointCloudDataFrame` for the reasons @mojaveazure elucidated elsewhere: `PointCloud` sounds like the points themselves, which might be some kind of n-dimensional array -- this is information _about_ them.

Reply:
That's more of a `somacore` conversation than a TileDB-SOMA decision.